### PR TITLE
The Secret Code Theme

### PR DESCRIPTION
**Pull request by busterbeam**
This should work, needs changes to the regular square tiler, but it can be passed on to the UI

### CODE REVIEW

**Comment by busterbeam** [comment]
last commit failing due to incorrect `type` for one of the function parameters/return. once fixed can be closed

**Comment by LukeHankey** [comment]
The tiles are not scrambled on start up like they were previously because the tile scrambler is not called. Is this intentional?

**Comment by busterbeam** [comment]
> The tiles are not scrambled on start up like they were previously because the tile scrambler is not called. Is this intentional?

Yes! before a scrambled image would be spat out by `regular` but instead I've separated the function out.  the scrambling function isn't that good so it is probably best that it's separated as that bit of code is going to get longer.

It's needed for secret_codes to work

File: jigsaw-game/common/secret_codes.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    text_parameters[0] = ImageFont.truetype(FONT_NAME, font_size)
+    draw = ImageDraw.Draw(image)
+    draw.text((0, 0), text, "white", *text_parameters)
+    return image
+
+
+def filter_tiles(
+    text_tiles: list[tuple[tuple[int, int], Image.Image]],
+    normal_tiles: list[tuple[tuple[int, int], Image.Image]],
+):
```
Comment (LukeHankey) [inline jigsaw-game/common/secret_codes.py:73]: Add a return type here

**Review by LukeHankey — CHANGES_REQUESTED** [review]
As above.

File: jigsaw-game/common/secret_codes.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        elif split_string[right_offset] == " ":
+            split_string[right_offset] = "\n"
+            break
+        right_offset -= 1
+        left_offset += 1
+    return "".join(split_string)
+
+
+def fitted_text_mask(
+    image: Image.Image, text: str, text_color: tuple[int, int, int] = (0, 0, 0)
```
Comment (LukeHankey) [inline jigsaw-game/common/secret_codes.py:29]: No need for `text_color` as doesn't seem to be used in the function.

File: jigsaw-game/common/secret_codes.py (hunk 1 above)
Comment (LukeHankey) [inline jigsaw-game/common/secret_codes.py:73]:
```suggestion
) -> Generator[tuple[tuple[int, int], Image | None], None, None]:
```

**Comment by LukeHankey** [comment]
You still need to import `Generator` from `collections.abc`. The module error type is coming from the return type you just added, where `Image` should be `Image.Image`

**Review by LukeHankey — APPROVED** [review]

**Pull request merged**